tests/src/lib.rs

```diff
@@ -21,6 +21,7 @@ pub static EXTRA_LINK: LazyLock<HashMap<&'static str, Vec<ExtraLinkage>>> = Lazy
         "loopy_funcs_not_inlined_by_default.c",
         "not_loopy_funcs_inlined_by_default.c",
         "reentrant.c",
+        "shadow_reentrant.c",
```

Comment:
I don't immediately see why this test needs extra linking?

Reply:
It calls to an external function `call_callback()` which is compiled externally.

Reply:
Ah!